kore/src/Kore/Simplify/Data.hs

```diff
@@ -20,17 +20,20 @@ module Kore.Simplify.Data (
 
     -- * Re-exports
     MonadSimplify (..),
```

Comment:
From the list above `-- * Re-exports`, only `evalSimplifier*` still lives in this module. Also, there isn't any "Data" left in `Simplify.Data` (maybe a misnomer in the first place).
Suggest we
1) either move all content from `Kore.Simplify.Data` to `Kore.Simplify`,
2) or make this a new "API" module that contains the runner and utility functions
Blast radius of 2) is of course bigger but once we move things around there is no reason to not do a proper job.

Reply:
By making this a new "API" module you mean renaming the 'Data' module `Kore.Simplify.API` and fixing the respective modules that import this one?

Reply:
I agree with @jberthold, we should probably do what he describes in option 1.

Reply:
@jberthold mentioned moving to `Kore.Simplify`, but there is no such module. Does that mean we need to create this module or was it a typo (was the actual module name `Kore.Simplify.Simplify`)?

Reply:
Sorry, oversight on my end... of course I mean `Kore.Simplify.Simplify`

Reply:
Oh, unfortunately, it seems like numerous circular imports emerge when we try to move everything from `Data` to `Simplify`.
In `Kore.Simplify.Data` we import `Kore.Simplify.TermLike`, and then if we import it in `Kore.Simplify.Simplify`, those tens of circular imports, written in the _Scope_ section of the pull request, make their appearance again.

Should we consider @jberthold's second option instead?

Reply:
Hmm, it's something that will have to be done in order to keep things clean, but I'm not sure how much time it will take. Maybe give it a try and let us know if it ends up being too out of scope for this issue?